Add DNS fields to Container and Network unit groups

ryanhockstad opened the pull request:
<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Quadlet unit files allow DNS field in Network group and DNS, DNSSearch, and DNSOption field in Container group.
```

closes #19884

TomSweeneyRedHat commented:
Quick late night browse LGTM, but definitely want less tired eyeballs on it.

ygalblum commented:
Code LGTM, but the linter is complaining on trailing-whitespace in docs/source/markdown/podman-systemd.unit.5.md

Luap99 commented on pkg/systemd/quadlet/quadlet.go:
```diff
@@ -483,6 +490,21 @@ func ConvertContainer(container *parser.UnitFile, names map[string]string, isUse
+
+	dnsOptions := container.LookupAll(ContainerGroup, KeyDNSOption)
+	for _, dnsOption := range dnsOptions {
+		podman.addf("--dns-search=%s", dnsOption)
```
typo this must be --dns-option

ryanhockstad replied:
Nice catch. Fixed this along with the trailing whitespace in the docs.

Luap99 commented:
Your new tests are failing

edsantiago commented:
@ryanhockstad please rebase on 9554022ce to get CI working.

mheon commented:
Code LGTM

rhatdan commented:
/lgtm
/hold
Thanks @ryanhockstad

ryanhockstad commented:
I think I fixed the tests I added that were failing. Should I squash the commit that merged the upstream into my branch? I sync'd my repo through the UI instead of rebasing.

rhatdan commented:
/approve

rhatdan commented:
/hold cancel
